cluster/engine.go

```diff
@@ -273,7 +273,7 @@ func (e *Engine) updateContainer(c dockerclient.Container, containers map[string
 	if full {
 		info, err := e.client.InspectContainer(c.Id)
 		if err != nil {
-			return nil, err
+			return containers, err
```

Comment:
I think we should fix the caller instead.

In Go, it is expected that if `err` is not nil then the rest of the response should be considered garbage.